src/app/DeviceSelector.tsx

```diff
@@ -59,9 +62,10 @@ const deviceListing: DeviceTraits = {
  * Note that the callbacks releaseCurrentDevice and onDeviceIsReady
  * are only invoked, if a deviceSetup is defined.
  */
-const onDeviceSelected = (dispatch: AppDispatch) => (device: Device) => {
+const onDeviceSelected = (dispatch: AppDispatch) => async (device: Device) => {
     logger.info(`Selected device with s/n ${device.serialNumber}`);
-    getBoardControllerVersion(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
+    await getBoardControllerVersion(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
+    await getCurrentBoardControllerConfig(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
```

Comment:
Confused about the FIXMEs here
Is this still to be done in the future or?

Reply:
I'm not quite sure, but I think so.